### ISSUE TITLE
 Standard names: river discharge and storage

### ISSUE THREAD

**Issue by DikraaK** [cf-convention/discuss#134]
**Proposer's name** Dikra Khedhaouiria
**Date** 2021-10-06

**- Term** temporal_mean_river_discharge
**- Description** The river_discharge refers to the volume of water flowing through a river channel and temporal_mean corresponds to the averaged river_discharge over the specified forecast_period 
**- Units** m<sup>3</sup>.s<sup>-1</sup>

**- Term** river_channel_storage
**- Description**The river_channel represents a body of water flowing along a natural channel and storage refers to the amount of water stored in that channel
**- Units** m<sup>3</sup>

**Comment by taylor13** [comment]
I think you could use the existing standard_name "water_volume_transport_in_river_channel" (canonical units m^3 s-1) and attach to the variable a cell_methods="time: mean" attribute (where time would be defined as the time coordinate).  You would then define the averaging period by attaching a "bounds" attribute to time (containing the beginning and ending times of the averaging period).

**Comment by feggleton** [comment]
@DikraK could you address the most recent comment from @taylor13

**Comment by egaborit** [comment]
> Would it be better to express it as a mass, rather than a volume? I'm thinking that river_channel_storage could possibly be confused with river_channel_capacity, but maybe I shouldn't be. Also, mass would better characterize things if we were to distinguish (using different standard_names) between frozen water and liquid water (since ice density varies). In any case, the definition of the standard_name should explicitly say whether the quantity includes both liquid and solid water or not.

For now, we would prefer to maintain the units of m^3 for channel storage. However, it is true that we should emphasize in the description that neither ice nor salinity are simulated by the model for now, so only unsalted liquid water is assumed 100% of the time everywhere for the channel storage variable. 

Regarding the possible confusion with channel capacity, I don't think it is an issue. However, we should also emphasize that channel storage can go beyond the main channel capacity, in which case the volume in excess of channel capacity is located in the "overbank" part of the channel.

**Comment by feggleton** [comment]
Thanks both, would this description cover that potential confusion? @taylor13 @egaborit @DikraK 

- Term: river_channel_storage
- Description: The 'river_channel' represents a body of water flowing along a natural channel and 'storage' refers to the amount of water stored in that channel. The quantity includes unsalted liquid water.
- Units m3

There are other phrases in other names which refer to liquid water but none that I have found which quite fit the bill.

**Comment by JonathanGregory** [comment]
Dear all

Similarly to Karl @taylor13, I feel it would be better if the standard name itself was more explicit about what it means (as we generally attempt to arrange with standard names). It's good, of course, to have a clear description as well, but better if we don't have to rely on it for essential information. Would you be happy with `water_volume_in_river_channel`, for example? That would correspond well to the existing name of `water_volume_`**`transport`**`_in_river_channel` (`m3 s-1`). "Storage" strikes me as a possibly confusing word, unless it really means the water is stagnant; it makes it sound like a reservoir, but I suppose the water is generally moving along the channel.

Best wishes

Jonathan

**Comment by egaborit** [comment]
@feggleton, @JonathanGregory : I agree with both of your last comments. 

However, note that in our WATROUTE gridded outputs, the volume of water in a given grid cell can either correspond to the volume of water contained in a river channel at a given time, or to the volume of water contained in a natural lake or in a managed reservoir, if the grid cell corresponds to the lake or reservoir outlet.

So it seems difficult to avoid the confusion between "stored" and "flowing" water here, without giving a longer description.

And by the way, I don't know to what variable the name "water_volume_transport_in_river_channel (m3 s-1)" refers to, 
but according to me, it would be better to use the term "water_flux_in_river_channel".

**Comment by dblodgett-usgs** [comment]
I agree that `storage` is the wrong term. The Unesco glossary and HY_Features treat storage as a term reserved for usable storage as in a reservoir. https://docs.ogc.org/is/14-111r6/14-111r6.html#_storage_of_water 

_containment_ is the concept I most frequently see for what is meant here. The river channel is a container for a volume of water and that measure is an instantaneous quantity within some pre-defined control volume. 

`water_volume_contained_in_river_channel` (m^3) may be a good way go? 

With regards to flow, it's important to recognize that the quantity is relative to a _specific control section_ or is _spatially averaged over a flowpath joining two tributaries_. In the hydrology domain, we would typically use the terms `discharge`, `convey`,  or `streamflow`. In this context, I like thinking of the channel as a _conveyance_ -- perhaps

`water_volume_conveyed_in_river_channel` (m^3/s) ??

**Comment by egaborit** [comment]
Maybe, but "discharge" and "streamflow" are indeed way more popular than the term "conveyed volume".

**Comment by taylor13** [comment]
I think ``water_volume_contained_in_river_channel`` and ``water_volume_in_river_channel`` would both be clear to experts and to those less familiar.   

I think "water flux in river channel" would have to have units of kg m-2 s-1, so an alternative should be found.

**Comment by JonathanGregory** [comment]
We already have `water_volume_transport_in_river_channel` in `m3 s-1`. Is that not what you mean, Dave @dblodgett-usgs? `transport` is a word we use for this kind of concept in other standard names (i.e. what would probably be called a flux in physics). If it's not an obvious word for some users, the obvious words (such as "streamflow" and "convey", which you mention) could be added to the description, to help someone searching. Jonathan

**Comment by egaborit** [comment]
Sounds good to me.

**Comment by dblodgett-usgs** [comment]
It would be helpful to add those keywords to the description. I agree that if we have a precedent for `water_volume_transport_in_river_channel` then `water_volume_contained_in_river_channel` is a good choice.

**Comment by efisher008** [comment]
Hi all, 

Many thanks for the useful comments, which I have . I have made a term change from `river_channel_storage` to `water_volume_contained_in_river_channel` in the CF editor, as this seemed to be the outcome of the discussion. 

Link to editor entry: [https://cfeditor.ceda.ac.uk/proposal/4758/edit](https://cfeditor.ceda.ac.uk/proposal/4758/edit)

Should there be any update to the [description suggested by @feggleton](https://github.com/cf-convention/discuss/issues/134#issuecomment-1759444291)? @JonathanGregory mentioned including _streamflow_ and _convey_ in the text. 

Best wishes (and happy Easter/Ramadan),
Ellie
